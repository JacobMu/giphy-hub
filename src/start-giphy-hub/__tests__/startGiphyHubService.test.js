import {
    getReviewCommentFormField,
    hasCommentElement,
    renderGiphyButton,
} from '../startGiphyHubService';

describe('startGiphyHubService', () => {
    describe('#hasCommentElement', () => {
        it('finds a element with class name js-reviews-toggle', () => {
            const querySelectorSpy = jest.spyOn(document, 'querySelector').mockReturnValue(true);

            const hasFoundElement = hasCommentElement();

            expect(querySelectorSpy).toHaveBeenCalledWith('.js-reviews-toggle');
            expect(hasFoundElement).toBe(true);
        });

        it('does not find a element with class name js-reviews-toggle', () => {
            const querySelectorSpy = jest.spyOn(document, 'querySelector').mockReturnValue(false);

            const hasFoundElement = hasCommentElement();

            expect(querySelectorSpy).toHaveBeenCalledWith('.js-reviews-toggle');
            expect(hasFoundElement).toBe(false);
        });
    });

    describe('#getReviewCommentFormField()', () => {
        document.body.innerHTML = '<div id="pull_request_review_body" />';
        it('returns PR Review Comment Form DOM element if exists', () => {
            const reviewElement = getReviewCommentFormField();

            expect(reviewElement['0'].outerHTML).toEqual(
                '<div id="pull_request_review_body"></div>',
            );

            document.body.innerHTML = '';
        });
    });

    describe('renderGiphyButton', () => {
        it('returns looks good to me html button template', () => {
            const button = renderGiphyButton();

            expect(button).toMatchSnapshot();
        });
    });
});
