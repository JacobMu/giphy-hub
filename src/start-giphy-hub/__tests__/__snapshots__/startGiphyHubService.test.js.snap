// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`startGiphyHubService renderGiphyButton returns looks good to me html button template 1`] = `
"
        <div id=\\"button-wrapper\\" class=\\"giphy-button-wrapper\\">
            <div id=\\"looks-good-to-me\\">LGTM</div>
        </div>
    "
`;
